compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/InnerClassesLowering.kt

```diff
@@ -28,6 +30,7 @@ import org.jetbrains.kotlin.ir.util.transformDeclarationsFlat
 import org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid
 import org.jetbrains.kotlin.ir.visitors.transformChildrenVoid
 
+// This pass has to run after LocalDeclarationsLowering, since we don't handle nested functions.
```

Comment:
Now, instead of documenting the dependency in a comment, you could specify it in the `prerequisite` field of the phase.

Reply:
The problem is that this particular lowering is run with different phases in the different backends (localDeclarationsLoweringPhase in JS, jvmLocalDeclarationsPhase in JVM, etc.), so I cannot point at a specific phase.

Reply:
I see. We will need to unify those phases at some point.

Reply:
But this variable (`innerClassesPhase`) is used only in JVM IR, could we move it to `JvmLower.kt` for example and specify the requirement on `jvmLocalDeclarationsPhase` there?

(Maybe we need to actually move all other JVM phase declarations from backend.common to backend.jvm exactly to be able to specify JVM-specific phase dependencies.)

Reply:
That's a good point, I didn't notice that the phase wasn't shared. I've added the prerequisite as part of #2342.